install/macos_packages/build_component_package.sh

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# Requirements:
+#  - MacOS
+#  - A starship repository with binaries and documentation already built.
+# Usage: run this script, passing $1 to the repository path. The script assumes
+# it is being run from within a starship repository if $1 is not provided.
+
+usage(){
+    echo "Builds a component package for macOS."
+    echo "Assumes that the following items already exist:"
+    echo "    - A starship binary which has already been notarized"
+    echo "    - Documentation created by \`npm run build\`, usually in a dist"
+    echo "      directory at <repo>/docs/.vuepress/dist"
+    echo "Usage: $0 <path-to-starship-binary> <path-to-dist-directory>"
+}
+
+script_dir="$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+source "$script_dir/common.sh"
+
+cleanup_server(){
+    if [[ -n "${server_pid-}" ]]; then
+        echo "Killing HTTP server ($server_pid) to clean up."
+        kill "$server_pid"
+        rm "x86_64-apple-darwin-simple-http-server"
+    else
+        echo "No server found, exiting normally."
+    fi
+}
+
+if [[ "$OSTYPE" != 'darwin'* ]]; then
+    error "This script only works on MacOS"
+fi
+
+if [[ "${2-undefined}" = "undefined" ]]; then
+    usage
+    exit 1
+fi
+
+starship_program_file="$1"
+starship_documentation_dir="$2"
+
+if [ ! -f "$starship_program_file" ]; then
+    error "Could not find starship binary at $starship_program_file"
+fi
+
+if [ ! -d "$starship_documentation_dir" ]; then
+    error "Could not find starship documentation at $starship_documentation_dir"
+fi
+
+pkgdir="$(mktemp -d)"
+mkdir -p "$pkgdir/usr/local/bin"
+cp "$starship_program_file" "$pkgdir/usr/local/bin/starship"
+
+# Now we get to make documentation! Vuepress was not designed to build locally
```

Comment:
Oof. If it must be done! 🙃

Reply:
Yeah, I was a little surprised that `npm serve` didn't actually yield an HTML hierarchy that could be scraped by wget. It seems to do the thing where you have a single line of JS which can be evaluated to generate the DOM for the real deal, but wget can't evaluate the JS, so it just pulls a single script.